docs/architetture-software/README.md

```diff
@@ -0,0 +1,211 @@
+- **Conoscenza del business**: comprendere gli obiettivi aziendali e i vincoli del progetto o prodotto è fondamentale. Queste figure devono allineare le loro decisioni tecniche con gli obiettivi strategici dell'organizzazione.
+- **Adattabilità**: il panorama del software è in continua evoluzione. La figura dovrebbe essere adattabile e aperta a nuove tecnologie e metodologie, nonché capace di rivalutare e adattare l’architettura quando necessario.
+
+## Ma quindi chi si occupa di architetture software deve essere necessariamente un senior?
+
+No, non necessariamente. Anche se è vero che il ruolo viene ricoperto da persone con una certa esperienza, non è detto che una figura junior non possa occuparsi di architetture software. Anzi, è molto importante che lo faccia, in quanto è un ottimo modo per imparare e crescere, ma il consiglio è di affiancare sempre una figura senior che possa guidare e supportare la persona junior.
+
+## Fitness functions
+
+Il termine _fitness function_ viene spesso adottato quando si parla di architetture evolutive, ovvero architetture software il cui scopo è quello di evolvere nel tempo, adattandosi ai cambiamenti del business e del mercato. Possono essere assimilabili a delle metriche che permettono di valutare quanto un'architettura software sia adatta a raggiungere gli obiettivi prefissati o ad eventuali obiettivi futuri del business.
+
+Apiumhub nel 2020 ha pubblicato un post sul proprio blog in cui definisce questo concetto in maniera molto chiara:
+
+> La _fitness functions_ è un tipo di funzione obiettivo utilizzata per riassumere quanto una determinata soluzione di architettura software sia vicina al raggiungimento degli obiettivi prefissati. Nel definire un'architettura evolutiva, l'architetto del software cerca un algoritmo "migliore"; la funzione fitness definisce cosa significa "migliore" in questo contesto.
+
+Esempi di fitness function possono essere: punteggio di scalabilità , percentuale di errori, tempo medio di risposta, quantità di messaggi persi dal broker, ecc.
+
+### Triggered Fitness functions
+
+Le _triggered fitness functions_ (funzioni di fitness _attivate_ o _triggerate_) vengono eseguite in base a un evento particolare, ad esempio il team di sviluppo che esegue un test unitario, una pipeline di distribuzione che esegue una serie di operazioni o una persona del QA che esegue determinate attività stabilite.
+
+Questo tipo di funzioni ha da un lato il vantaggio di poter essere eseguito in maniera molto rapida, ma dall'altro lato ha il difetto di non essere rappresentative di una situazione reale. Questo tipo di fitness function non tiene in considerazione, infatti, il carico di lavoro che il sistema deve gestire durante il _day-by-day_, ma solo situazioni isolate e controllate.
```

Comment:
Qui non mi hai convinto:

"Questo tipo di funzioni ha da un lato il vantaggio di poter essere **eseguito** in maniera molto rapida" => maschile singolare

"il difetto di non essere **rappresentative** di una situazione reale" => femminile plurale

Prendila con le pinze perché sono una capra, magari farei un passaggio con il @Il-Libro-Open-Source/drafting-group

Reply:
Hai ragione, corretto!